src/tool/xmeta/cpp/models/base_model.h

```diff
@@ -26,12 +31,65 @@ namespace xlang::xmeta
     struct method_model;
     struct event_model;
 
-    using class_type_semantics = std::variant<
+    struct formal_parameter_model;
+    struct symbol_table;
+    struct compilation_unit;
+    struct type_ref;
+
+    enum class compilation_error
+    {
+        passed = 0,
+        symbol_exists,
+        accessor_exists,
+        method_cannot_be_overloaded
+    };
+
+
+    enum class method_association
+    {
+        None,
+        Property,
+        Event,
+        Constructor
+    };
+
+    enum class simple_type
+    {
+        Boolean,
+        String,
+        Int8,
+        Int16,
+        Int32,
+        Int64,
+        UInt8,
+        UInt16,
+        UInt32,
+        UInt64,
+        Char16,
+        Single,
+        Double,
+    };
+
+    struct object_type {};
+
+    using class_type_semantics = std::variant<std::monostate,
```

Comment:
The meta_reader project refers to these ase type categories. And it seems odd to call them "class" type semantics, when only one of the possible variant options is class